main.go

```diff
@@ -83,6 +102,31 @@ func main() {
 	os.Exit(exitCode)
 }
 
+// Check status of SDS secrets from Istio to make sure SDS bootstrap was successful
+// This SDS data is populated at bootstrap Before the actual xDS connections are established so it should be safe to
+// just kill the envoy pod qith `quitquitquit` if we are in a bad state here
+func checkEnvoyIstioSDS(host string) error {
+	url := fmt.Sprintf("%s/certs", host)
+
+	rsp := typhon.NewRequest(context.Background(), "GET", url, nil).Send().Response()
+
+	certs := &Certificates{}
+
+	err := rsp.Decode(certs)
+	if err != nil {
+		return err
+	}
+
+	// Check number of CERTS is > 1, if not hit the KILL Endpoint
+	if len(certs.Certificates) == 1 {
```

Comment:
Perhaps `len(certs.Certificates) < 2` ? to catch the 0 case? I assume that is also an error?

Reply:
done